src/components/partials/Book.js

```diff
@@ -0,0 +1,30 @@
+/*eslint-disable */
+
+
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { actionRemoveBook } from '../../redux/books/books';
+
+const Book = (props) => {
+  const { book } = props;
+  const dispatch = useDispatch();
+  return (
+    <li>
+      <p>{book.title}</p>
+      <p>{book.author}</p>
+      <button type='button' onClick={() => {dispatch(actionRemoveBook(book))}}>Remove</button>
+    </li>
```

Comment:
- [ ] After adding several books to the store, they get added and rendered to the screen but when I click on the remove button, it removes the entire books added. You should pass the book id into the `actionRemoveBook` action creator instead of the book object as your payload in the action creator is receiving payload.id. Kindly work on that so one can only delete a single book after adding several books and not remove all books added at once. 👍 

[ OPTIONAL CHANGE ]
- It is good to keep your JSX less logical and extract the direct function call in the remove button into a removeBookFromStore function which you can call on the button. That will make your JSX clean and not too busy handling logic.

Reply:
Hello @AmaduKamara, I would really love to implement this optional change but am not understanding it.

Please note that actionRemoveBook is an action creator.

Thank you for your advice.

Reply:
Hey @wayungi, I know it's an action creator.

What I mean is for you to create a separate function, and then inside that function, you do the dispatch there just as you did it inside this function in the button. Doing that will make your JSX code lean and not crowded with logic.

Reply:
```mdx
const removeBookFromStore = () => {
    dispatch(actionRemoveBook(book.id));
  };
```
And your button should look like this:
```mdx
<button type="button" onClick={removeBookFromStore}>
        Remove
      </button>
```